charms/worker/k8s/src/charm.py

```diff
@@ -577,51 +595,76 @@ def _reconcile(self, event: ops.EventBase):
             self._copy_internal_kubeconfig()
             self._expose_ports()
 
-    @on_error(
-        ops.WaitingStatus("Cluster not yet ready"),
-        AssertionError,
-        subprocess.CalledProcessError,
-        InvalidResponseError,
-        K8sdConnectionError,
-    )
     def _update_status(self):
         """Check k8s snap status."""
-        if self.is_dying:
-            status.add(ops.WaitingStatus("Preparing to leave cluster"))
-            return
-        if self.is_control_plane:
-            assert self.api_manager.is_cluster_ready(), "control-plane not yet ready"  # nosec
-
-        if version := self._get_snap_version():
+        if version := snap_version("k8s"):
             self.unit.set_workload_version(version)
 
+        if not self.get_cluster_name():
+            status.add(ops.WaitingStatus("Node not Clustered"))
+            return
+
         if not self._is_node_ready():
-            status.add(ops.WaitingStatus("Node not yet Ready"))
+            status.add(ops.WaitingStatus("Node not Ready"))
             return
```

Comment:
If the node is IN the cluster, but has a `NotReady` status (or it can't be found b/c the api server is down), they'll see this